torchaudio_contrib/functional.py

```diff
@@ -118,107 +145,76 @@ def create_mel_filter(n_stft, sr, n_mels=128, f_min=0.0, f_max=None):
     up_slopes = slopes[:, 2:] / f_diff[1:]  # (n_stft, n_mels)
     fb = torch.clamp(torch.min(down_slopes, up_slopes), min=0.)
 
-    return fb, f_pts[:-2]
-
+    return fb
 
-def amplitude_to_db(spec, ref=1.0, amin=1e-10, top_db=80.0):
-    """
-    Amplitude spectrogram to the db scale
-
-    Input Tensor shape -> (freq, time)
-    Output Tensor shape -> (freq, time)
-    """
-    power = spec**2
-    return power_to_db(power, ref, amin, top_db)
-
-
-def power_to_db(spec, ref=1.0, amin=1e-10, top_db=80.0):
-    """
-    Power spectrogram to the db scale
 
-    Input Tensor shape -> (freq, time)
-    Output Tensor shape -> (freq, time)
+def melspectrogram(x, n_mels=128, sr=44100, f_min=0.0, f_max=None, **kwargs):
     """
-    if amin <= 0:
-        raise ParameterError('amin must be strictly positive')
-
-    if callable(ref):
-        ref_value = ref(spec_norm)
-    else:
-        ref_value = torch.tensor(ref)
-
-    log_spec = 10*torch.log10(torch.clamp(spec, min=amin))
-    log_spec -= 10*torch.log10(torch.clamp(ref_value, min=amin))
-
-    if top_db is not None:
-        if top_db < 0:
-            raise ParameterError('top_db must be non-negative')
-        log_spec = torch.clamp(log_spec, min=(log_spec.max() - top_db))
-
-    return log_spec
-
-
-def db_to_power(spec_db, ref=1.0):
+    Compute the melspectrogram of a given signal.
+    See torchaudio_contrib.Melspectrogram for more details.
     """
-    db-scale spectrogram to power spectrogram
+    spec = spectrogram(x, **kwargs)
+    fb = create_mel_filter(
```

Comment:
Again, I'm quite sure the filterbank should be a parameter to the function, not created inside.

Reply:
Touched on it in the [first comment](https://github.com/keunwoochoi/torchaudio-contrib/pull/34#discussion_r271553604): if filterbank is passed as a parameter to the function then so is the spectrogram (or at least its window), then maybe we're not too interested in `functional.melspectrogram` since it becomes a wrapper for mm? Not that that's bad but I think we should discuss the role of `functional`!